util/fibers/uring_proactor.cc

```diff
@@ -325,6 +347,62 @@ void UringProactor::ReturnRegisteredBuffer(uint8_t* addr) {
   free_req_buf_id_ = buf_id;
 }
 
+int UringProactor::RegisterBufferRing(unsigned group_id) {
+  if (buf_ring_f_ == 0)
+    return EOPNOTSUPP;
+
+  if (bufring_groups_.size() <= group_id) {
+    bufring_groups_.resize(group_id + 1);
+  }
+
+  auto& ring_group = bufring_groups_[group_id];
+  CHECK(ring_group.ring == nullptr);
+
+  int err = 0;
+
+  ring_group.ring = io_uring_setup_buf_ring(&ring_, kBufRingEntriesCnt, group_id, 0, &err);
+  if (ring_group.ring == nullptr) {
+    return -err;  // err is negative.
+  }
+
+  unsigned mask = kBufRingEntriesCnt - 1;
+  ring_group.buf = new uint8_t[kBufRingEntriesCnt * kBufRingEntrySize];
```

Comment:
Does this buffer need to be aligned for any specific type?

Reply:
not that I know of. it's a binary blob that is used to read the data from socket to.